src/Xamarin.Android.Build.Tasks/Utilities/ApplicationConfigNativeAssemblyGenerator.cs

```diff
@@ -16,8 +21,77 @@ enum MonoComponent
 		Tracing   = 0x04,
 	}
 
-	class ApplicationConfigNativeAssemblyGenerator : NativeAssemblyGenerator
+	class ApplicationConfigNativeAssemblyGenerator : NativeAssemblyComposer
```

Comment:
Possible crazy talk, but should we change the suffix if this & other classes from `Generator` to `Composer` to match the new base class?  Why did the base class name change?

Reply:
Because the classes that derive from `NativeAssemblyComposer` no longer have any code that directly generates the assembler code. They **compose** the shape of the assembly source file, giving it structure, and call high-level methods on `NativeAssemblyGenerator` to generate actual code (or output strings, assign labels etc). They no longer deal with any low-level **assembler** code output directly. I didn't rename them because from the "user's" point of view (that is, the Tasks that call the `NativeAssemblyComposer` derivatives) the `ApplicationConfigNativeAssemblyGenerator` is **still** a generator that outputs assembly code. The base class of `ApplicationConfigNativeAssemblyGenerator` is a technicality that's of no interest to the caller, and the `Generator` suffix describes function of the class, not its place in some hierarchy which is an internal design detail.

The goal of the redesign was to make modifying the `*Generator` classes easier for people not familiar with native assembly.